### PR TITLE
2720 - Definitive Global/Country file fix, support for UTF-8 filenames

### LINKED ISSUE #2720: Garbled file name in Links & Repository
@minotogna 

The file name is garbled and not sure if this file should appear in Links & Repository for all the countries since it is only in arabic.....

### PR DESCRIPTION
**Pull request by rfontanarosa**
close #2720

upload: multer, due to busboy dependency, doesn't support utf-8 filenames by default. there is a workaround to fix it: https://github.com/expressjs/multer/issues/1104

download: nodejs doesn't support utf-8 on response header content-disposition: https://stackoverflow.com/questions/23895924/expressjs-download-filename-utf-8

primary key: pgadmin doesn't allow me to change file table on the fly because it hadn't a primary key. this problem can occurs with others sql clients. I just added a primary key as in the other tables (col, row)

https://github.com/openforis/fra-platform/assets/10047945/1a35b411-be16-4691-b221-214a3ece6660

### CODE REVIEW

**Review by sorja — DISMISSED** [review]

**Comment by sorja** [comment]
wrong video?

**Comment by Doc999tor** [comment]
There is a PR for Multer that solves the issue - https://github.com/expressjs/multer/pull/1210

File: src/test/migrations/steps/20230607170529-step-primary-key-for-table-file.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { BaseProtocol } from 'server/db'
+
+export default async (client: BaseProtocol) => {
+  await client.query(`ALTER TABLE assessment_fra.file ADD PRIMARY KEY (id);`)
+
+  await client.query(`ALTER TABLE assessment_paneuropean.file ADD PRIMARY KEY (id);`)
```
Comment (minotogna) [inline src/test/migrations/steps/20230607170529-step-primary-key-for-table-file.ts]: I think also the unique uuid added to the ddl must be included here ?

**Review by sorja — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Comment by pengchenglai** [comment]
@minotogna 
@sorja 
@rfontanarosa 

Great to know it is fixed, plz move this file([Oportunidades de restauracion en Guatemala.pdf](https://fra-data.fao.org/api/file/assessment/8f42a366-5b26-4cb8-86ff-9c7cb276f335?countryIso=X18&assessmentName=fra&cycleName=2025)) to the **Repository of Guatemala**
Capture

Remove other two testing files:

**Pull request merged**